Modules/Log/Get-RegistryKeyValData.ps1

```diff
@@ -0,0 +1,169 @@
+			[UInt32 ].MakeByRefType(), #Value Count
+			[UInt32 ].MakeByRefType(), #Max Value Name Length
+			[UInt32 ].MakeByRefType(), #Max Value Name Length
+			[UInt32 ].MakeByRefType(), #Security Descriptor Size           
+			[long].MakeByRefType() #LastWriteTime
+		) #Method Parameters
+	)
+
+	$DllImportConstructor = [Runtime.InteropServices.DllImportAttribute].GetConstructor(@([String]))
+	$FieldArray = [Reflection.FieldInfo[]] @(       
+		[Runtime.InteropServices.DllImportAttribute].GetField('EntryPoint'),
+		[Runtime.InteropServices.DllImportAttribute].GetField('SetLastError')
+	)
+
+	$FieldValueArray = [Object[]] @(
+		'RegQueryInfoKey', #CASE SENSITIVE!!
+		$True
+	)
+
+	$SetLastErrorCustomAttribute = New-Object Reflection.Emit.CustomAttributeBuilder(
+		$DllImportConstructor,
+		@('advapi32.dll'),
+		$FieldArray,
+		$FieldValueArray
+	)
+
+	$PInvokeMethod.SetCustomAttribute($SetLastErrorCustomAttribute)
+	#endregion RegQueryInfoKey Method
+
+	[void]$TypeBuilder.CreateType()
+	#endregion DllImport
+} ## End of Name: Get-RegistryKeyTimestamp
+
+# Test to make sure the provide key is valid.
+Try{
+	# If valid, get property and set RegistryKey value
+	If (test-path -Path registry::$Key) {
+		Get-ItemProperty -Path Registry::$Key
+		$RegistryKey = Get-Item -Path Registry::$Key
+
+		## The following code is from
+		## Name: Get-RegistryKeyTimestamp
+		##		Author: Boe Prox
+		##		Version History:
+		##			1.0 -- Boe Prox 17 Dec 2014
+		##				-Initial Build
+		#region Constant Variables
+		$ClassLength = 255
+		[long]$TimeStamp = $null
+		#endregion Constant Variables
+
+		$ClassName = New-Object System.Text.StringBuilder $RegistryKey.Name
+		$RegistryHandle = $RegistryKey.Handle
+		#endregion Registry Key Data
+
+		#region Retrieve timestamp
+		$Return = [advapi32]::RegQueryInfoKey(
+			$RegistryHandle,
+			$ClassName,
+			[ref]$ClassLength,
+			$Null,
+			[ref]$Null,
+			[ref]$Null,
+			[ref]$Null,
+			[ref]$Null,
+			[ref]$Null,
+			[ref]$Null,
+			[ref]$Null,
+			[ref]$TimeStamp
+		)
+		Switch ($Return) {
+			0 {
+				#Convert High/Low date to DateTime Object
+				$LastWriteTime = [datetime]::FromFileTime($TimeStamp)
+
+				#Return object
+				$Object = [pscustomobject]@{
+					FullName = $RegistryKey.Name
+					Name = $RegistryKey.Name -replace '.*\\(.*)','$1'
+					LastWriteTime = $LastWriteTime
+				}
+				$Object.pstypenames.insert(0,'Microsoft.Registry.Timestamp')
+				$Object
+			}
+			122 {
+				Throw "ERROR_INSUFFICIENT_BUFFER (0x7a)"
+			}
+			Default {
+				Throw "Error ($return) occurred"
+			}
+		}
+		#endregion Retrieve timestamp
+		## End of Name: Get-RegistryKeyTimestamp
+	}
+	# if not, write error message
+	else {
+		Write-Error -Message "Key does not exist"
+	}
+}
+
+# If for some reason everything fails, write error message
+Catch{
+	Write-Error -Message "Unable to retrieve data from registry"
+}
+
+if ($Error) {
+	# Write the $Error to the $Errorlog
+    Write-Error "Get-RegistryKeyValData Error on $env:COMPUTERNAME"
```

Comment:
You shouldn't need to explicitly add the target's computer name. Error handling in the main script already takes care of that.